grails-app/assets/javascripts/spApp/directive/selectSpecies.js

```diff
@@ -192,21 +232,42 @@
                                     var pos = query.q.indexOf(scope.filters[i])
                                     if (pos >= 0) query.q.splice(pos, 1)
                                 }
-
-                                // apply spatial validity options
-                                var includeTrue = scope.spatiallyValid;
-                                var includeFalse = scope.spatiallySuspect;
-                                var includeNull = scope.spatiallyUnknown;
-                                var gs = ["-*:*"];
-                                if (includeTrue && !includeFalse) {
-                                    if (includeNull) gs = ["-geospatial_kosher:false"]
-                                    else gs = ["geospatial_kosher:true"]
-                                } else if (!includeTrue && includeFalse) {
-                                    if (includeNull) gs = ["-geospatial_kosher:true"]
-                                    else gs = ["geospatial_kosher:false"]
-                                } else if (includeTrue && includeFalse) {
-                                    if (includeNull) gs = []
-                                    else gs = ["geospatial_kosher:*"]
+                                /*
+                                   spatially-valid = geospatial_kosher:true
+                                   spatially-suspect = geospatial_kosher:false
+                                   spatially-unknown = -geospatial_kosher:*
+
+                                   If we want include VALID and MISSING(UNKNOWN) spatial data
+                                   fq=(geospatial_kosher:true OR -geospatial_kosher:*) BS(solr) does not support '-' in complicated query
+                                   */
+                                var gs = ["-*:*"]; // select nothing
+                                if (scope.spatiallyUnknown) { //include UNKNOWN (MISSING) spatial data records
+                                    if (scope.spatiallyValid && scope.spatiallySuspect) {  //All selected
+                                        //Returns all records
+                                        gs = ["*:*"]
+                                    } else if (scope.spatiallyValid) {
+                                        //  spatially-unknown && spatiallyValid
+                                        //  Solution -> rule out of spatiallySuspect
+                                        gs = ['-geospatial_kosher:false'];
+                                    } else if (scope.spatiallySuspect) {
+                                        //  spatially-unknown && spatiallySuspect
+                                        //  -> rule out of spatiallyValid
+                                        gs = ['-geospatial_kosher:true'];
+                                    } else {
```

Comment:
Not sure this `else` is needed? You didn't include it in version above ( grails-app/assets/javascripts/spApp/directive/selectFacet.js line 49). If you keep it in, then maybe it should throw an exception that the front-end can display a warning??? Otherwise the user will just get a wierd result they don't understand. I.e. no records appearing on the map - that will happen with `'-geospatial_kosher:*'`

Reply:
That was original thought - return empty.  Now, the front (GUI) prevents no selection on spatial options

Reply:
But we just fixed this bug where front-end was not supposed to allow this combination through (it used to prevent it) and the bug exposed this option. So if another bug is introduced in front end code, then user could get this combination again. If that happens, how will the user (or us) know the bug exists? We won't if it just returns empty, as we might just think "maybe there are no records for this species??" as opposed to thinking "OH I got an error message here - I'll send an email to support and let them know". We need to catch exceptions and propagate them to the UI so we know about it when it happens.